packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -971,7 +971,12 @@ export const ComboBoxMixin = (subclass) =>
           this._selectItemForValue(this.value);
         }
 
-        if (this._inputElementValue === undefined || this._inputElementValue === this.value) {
+        const inputValue = this._inputElementValue;
+        if (
+          inputValue === undefined ||
+          inputValue === this.value ||
+          inputValue === this._getItemLabel(this.selectedItem)
```

Comment:
The `selectedItemChanged` observer should have already set `_inputElementValue` to the selected item value by this time and the previous check should be enough. Why doesn't it work?

Reply:
When using object items, `value` property is set based on the `itemValuePath` property, whereas the input value is set based on the `itemLabelPath` - this is the case that wasn't covered previously.

Reply:
Ah, now I understand 👍 Then this part of the condition becomes redundant and can be removed:

```diff
inputValue === undefined ||
- inputValue === this.value ||
inputValue === this._getItemLabel(this.selectedItem)
```

It is indeed more correct to check that `inputValue` matches the selected item's label.

Reply:
Thanks, makes sense. Updated.